examples/run_rolling_forecast_backtest.py

```diff
@@ -0,0 +1,156 @@
+    print('ds_eval passes target length check\n')
+
+
+def generate_expected_predict():
+    a = []
+    for i in range(10):
+        for ii in range(5):
+            a.append((float(i),25 - ii - 1))
+
+    return a
+
+def test_ds_pred_targets(datasets):
+    ds_pred = iter(datasets['to_predict'])
+
+    to_compare = generate_expected_predict()
+    i = 0
+    for ts in ds_pred:
+        #print(len(ts['target']), to_compare[i][1])
+        assert len(ts['target']) == to_compare[i][1]
+        for val in ts['target']:
+            #print(val, to_compare[i][0])
+            assert val == to_compare[i][0]
+        i = i + 1
+    print("ds_pred passes target length and value checks\n")
+
+def generate_datasets():
+    dataset = get_dataset("constant", regenerate=False)
+    '''
+    estimator = SimpleFeedForwardEstimator(
+        prediction_length=dataset.metadata.prediction_length,
+        freq=dataset.metadata.freq,
+        trainer=Trainer(epochs=5, num_batches_per_epoch=10),
+    )
+    '''
+    #predictor = estimator.train(dataset.train)
+
+    # calculate time points of test set
+    ts = next(iter(dataset.test))
+
+    length = len(ts['target']) # 30
+    len_to_truncate = 5
+    length_of_roll = 5
+    test_end = length - len_to_truncate # 25
+    # we add one as 21, 22, 23, 24 and 25 will be the total items that will be in the rolling forecast
+    test_start = test_end - length_of_roll + 1  # 21
+    
+    print('test_start: {}\ntest_end: {}'.format(test_start, test_end))
+
+    # Calculate Timestamps of test range
+    t = ts['start']
+    f = t.freq
+    t_end = t + f * (length - 1) 
+    t0 = t + f * (test_start - 1)
+    t1 = t + f * (test_end - 1)
+
+    t_index = 0
+    t0_index = 20
+    t1_index = 24
+    t_end_index = 29
+
+    print("start: {}\nt0: {}\nt1: {}\nt_end: {}".format(t, t0, t1, t_end))
+    print("start: {}\nt0: {}\nt1: {}\nt_end: {}".format(t_index, t0_index, t1_index, t_end_index))
+
+    assert(t0_index == len(pd.date_range(start=t, end=t0, freq=f))-1)
+    assert(t1_index == len(pd.date_range(start=t, end=t1, freq=f))-1)
+    assert(t_end_index == len(pd.date_range(start=t, end=t_end, freq=f))-1)
+
+    # t0 is start of window to perform rolling forecasts on
+    # t1 is end of window to perform rolling forecasts on
+
+    datasets = generate_rolling_datasets (dataset.test, 1, t0, t1)
+    return datasets
+
+if __name__ == "__main__":
+
+    # test so that length of evaluation dataset is consistent with the predict dataset
+    test_dataset_lengths(generate_datasets())
+    test_ds_eval_targets(generate_datasets())
+    test_ds_pred_targets(generate_datasets())
+    
+    
+
+
+
+
+
+
+
+
+
+    exit()
+    '''
+    forecast_it, ts_it = make_evaluation_predictions(
+        dataset.test, predictor=predictor, num_samples=100, rolling_time_range=(t0,t1)
+    )
+
+    agg_metrics, item_metrics = Evaluator()(
+        ts_it, forecast_it, num_series=len(dataset.test)
+    )
+
+    pprint.pprint(agg_metrics)
+    '''
```

Comment:
Always have a new line at the end of the file!

Reply:
yep, missed that